### DISCUSSION TITLE
Security Status of Mesh Central post Ylian and team

### DISCUSSION THREAD

**Discussion by southeasterntech** [Ylianst/MeshCentral#6599] [Q&A]
Hi All, 

We have become quite dependent on Mesh for everything.  It's truly an incredible tool.  Some of our people have raised concerns lately that Mesh, being open source, might have code exposed and security issues.  

When Ylian and team were daily involved, I had no doubts in my mind.  I know Ylian said he would no longer be actively developing it, but rather making sure it stayed secure.

1.  When was the last comprehensive Pen test\vuln scan or security audit of Mesh?  
2.   What is the status of current security development
3.  Is there a 2024\2025 security hardening guide?  

Thanks!
Setech

**Comment by marclaporte** [comment]
We now have monthly community meetings, so please join us:
https://github.com/Ylianst/MeshCentral/wiki/Community-Monthly-Meetings

**Reply by NewUserfromDE** [reply]
> 3. Is there a 2024\2025 security hardening guide?

IMHO this third point a) is and b) is worth a separate thread. And although Marc is right to attend the community meeting for general security discussion this is good for points 1 and 2, but less for point 3, the hardening guide.

So as a reminder at last for me:
* there should be a hardening guide
* I shall discuss this on one of the community meetings
* maybe (re)post this some time in the future as a separate thread

From a strategic point of view IMHO (everything regarding) security is one crucial point for the (further) success of MeshCentral. Being open source (yeah!) creates problems commercial applications don't have, because open sources makes a software easily available:

* more abuse for scam and so on (reasons: free and more modifiable)
* more unsafe setups / more unprepared or uninformed admins (reasons: free(!) - and foreseeable future: turn-key installations and the like. So overall a lower barrier for an own remote access/management server.)

So I'm always attentive, when it comes to security and MeshCentral. And I had the same question (hardening guide) and therefore found this thread ;-)

Conclusion / Reminder: discuss creating a hardening guide

**Comment by southeasterntech** [comment]
Very much agree!  Shane D. LewisTechnical ***@***.***: 6788076156 x 701   ----- Original message ----- From: "NewUserfromDE" ***@***.***> To: "Ylianst/MeshCentral" ***@***.***> Cc: "Southeastern Technical" ***@***.***>, "Author" ***@***.***> Subject: Re: [Ylianst/MeshCentral] Security Status of Mesh Central post Ylian and team (Discussion #6599)Date: Wed, Jan 29, 2025 7:49 PM CAUTION: This email originated from outside the Southeastern network. Do not click links or open attachments unless you recognize the sender and know the content is safe. This warning was created by Southeastern Technical Information Security Services. Is there a 2024\2025 security hardening guide?IMHO this a) is and b) is worth a separate thread. And although Marc is right to attend the community meeting for general security discussion this is good for points 1 and 2, but less for point 3, the hardening guide.So as a reminder at last for me:there should be a hardening guideI shall discuss this on one of the community meetingsmaybe (re)post this some time in the future as a separate threadFrom a strategic point of view IMHO (everything regarding) security is one crucial point for the (further) success of MeshCentral. Being open source (yeah!) creates problems commercial applications don't have, because open sources makes a software easily available:more abuse for scam and so on (reasons: free and more modifiable)more unprepared or uninformed admins (reasons: free(!) - and foreseeable future: turn-key installations and the like)So I'm always attentive, when it comes to security and MeshCentral. And I had the same question (hardening guide) and therefore found this thread ;-)Conclusion / Reminder: discuss creating a hardening guide—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by J-Palomino** [comment]
![Alt Text](https://media.giphy.com/media/vFKqnCdLPNOKc/giphy.gif)
+1 for hardening guide, but is there a TODAY solution?
 would putting everything behind a vpn be a good start? 
 does that even matter considering meshcentrals chosen handshake protocol? 
 as long as we keep our private certs secret the connection between server and client is probably not the most vulnerable attack vector right? 
 So we should also be focused on security at the UI level since we are using node theres not much decoupling between model and view?
Am i getting any of this right?

**Comment by mwllgr** [comment]
Just as a starting point, the solutions mentioned at https://www.reddit.com/r/MeshCentral/comments/ob8o2w/howto_a_paranoid_guys_guide_to_protecting/ are still very much recommended!

I think one of the best things one can do is using the split agent and web UI port and making sure that the web UI itself is not reachable via the public web or only with decent security measures (for example secured by client certificate authentication with mTLS)!

**Comment by PrplHaz4** [comment]
@si458 any chance you can provide an update on how security posture is being maintained currently (for MeshCentral and MeshAgent)?  it may be worth a mention in the readme - the influx of AI-driven PRs and overall size of changesets makes this nearly impossible for a human to keep up with, and given the nature of the tool is probably a pretty high-profile target.

I saw CodeQL being run automatically but also some mentions of monthly security scans, but no real evidence of what's being done/found/fixed...etc.

**Reply by si458** [reply]
code is human reviewed, if i dont understand a PR i ask my LOCAL AI (not cloud) what the PR does, and it explains to me like a child
(yes i actually add that prompt into the ai `explain it to me like in a child!` and it actually does it good haha)
then i verify all is well or if any changes are needed and then merge when comepleted!

only Ylian can push to NPM (hense the problem at the moment with the 1.2.6 release as NPM wont push it/show it for some reason?)

AI is acceptable to use if you choose to use it, but you must SAY/DISCLOSE you used it (part of the PR template now)

ylian also setup the codeql to scan on every commit and pr so it tells us if the is problems etc
https://github.com/Ylianst/MeshCentral/actions/workflows/codeql-analysis.yml

dependabot also runs to let us know when a package inside of packaage.json needs updating and again this is done manually we dont let it just update because we have to test it all first etc]

hope this helps!

**Reply by PrplHaz4** [reply]
> code is human reviewed, if i dont understand a PR i ask my LOCAL AI (not cloud) what the PR does, and it explains to me like a child (yes i actually add that prompt into the ai `explain it to me like in a child!` and it actually does it good haha) then i verify all is well or if any changes are needed and then merge when comepleted!
> 
> only Ylian can push to NPM (hense the problem at the moment with the 1.2.6 release as NPM wont push it/show it for some reason?)
> 
> AI is acceptable to use if you choose to use it, but you must SAY/DISCLOSE you used it (part of the PR template now)
> 
> ylian also setup the codeql to scan on every commit and pr so it tells us if the is problems etc https://github.com/Ylianst/MeshCentral/actions/workflows/codeql-analysis.yml
> 
> dependabot also runs to let us know when a package inside of packaage.json needs updating and again this is done manually we dont let it just update because we have to test it all first etc]
> 
> hope this helps!

Thanks so much for the thoughtful, thorough response!  And thanks so much for doing the reviews - the volume of changes coming in makes that a tough task to do manually so glad you are getting at least a bit of AI help.

Please let the community know if/where help would be appreciated in this regard (even if it's as mundane as donating tokens or something!) - I'm sure many of us would like to help ensure this fantastic (and secure) product that Ylian and now you have built remains that way!
